### PR TITLE
Add scream dockerfile 

### PR DESCRIPTION
**Pull request by elynnwu**
This PR adds a dockerfile for eamxx. It is built on top of CIME's docker container with additional packages for eamxx and google cloud platform sdk for interfacing with AI2's ML workflow. Full model run and eamxx unit tests were completed inside the container and should be added to CI in the future.

### CODE REVIEW

File: components/eamxx/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            blas && \
+            rm -rf /opt/conda/pkgs/*
+
+# install gcloud
+RUN apt-get update && apt-get install -y  apt-transport-https ca-certificates gnupg curl gettext
+
+RUN echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list &&\
+    curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+
+RUN apt-get update && apt-get install -y google-cloud-sdk
```
Comment (elynnwu) [inline components/eamxx/docker/Dockerfile:22]: This if the official installation instruction from [Google Cloud CLI](https://cloud.google.com/sdk/docs/install-sdk#deb).

**Comment by PeterCaldwell** [comment]
Context: Elynn is working on my LDRD seeking to emulate SCREAM by adding a ML-based nudging process to low-res EAMxx. Her institute (AI2) uses the cloud for most of their computing,so being able to run CIME-based SCREAM runs in the cloud is important to her (and might be useful to us). This PR implements this capability.

**Comment by AaronDonahue** [comment]
@jgfouca , @ambrad and @bartgol I added you as reviewers thinking one of you may have an opinion on implementation here.  Feel free to take yourself on the reviewer list if you don't.

**Review by jgfouca — APPROVED** [review]
Nice! Looks good to me.

**Pull request merged**